packages/dds/matrix/src/matrix.ts

```diff
@@ -568,9 +592,18 @@ export class SharedMatrix<T extends Serializable = Serializable>
 
     protected async loadCore(branchId: string | undefined, storage: IChannelStorageService) {
         try {
-            await this.rows.load(this.runtime, new ObjectStoragePartition(storage, SnapshotPath.rows), branchId);
-            await this.cols.load(this.runtime, new ObjectStoragePartition(storage, SnapshotPath.cols), branchId);
-            const [cellData, pendingCliSeqData] = await deserializeBlob(this.runtime, storage, SnapshotPath.cells);
+            const serializer = this.runtime.IFluidSerializer;
```

Comment:
General observation - I'd try to remove this.runtime.IFluidSerializer and instead have instead in SharedObject.
Pri 1 reason - just to make sure that we inspected every usage and made determination that right object is used, i.e. we do not accidently use it in snapshots.
Pri 2 - that's not the right interface, DDS should use runtime.channelsRoutingContext (not all work is done in this area) to build its own serializer, the only thing on the interface between runtime and channel is exposer of routing / paths, that's it.

Reply:
I had actually done this in one of the iterations but changed it back. Let me see how big of a change this is; if its too big, I would rather do that in a separate PR because I feel it will have issues of its own.

Reply:
Added a serializer of its own to SharedObject. Removed references to DataStoreRuntime's IFluidSerializer.